ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -220,56 +230,81 @@ where
 where
     D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
     H: 'static + StorageHasher + Sync,
 {
     let mut changed_keys = BTreeSet::default();
-    let token = wrapped_erc20s::token(asset);
-    let balance_key = balance_key(&token, receiver);
-    update::amount(wl_storage, &balance_key, |balance| {
-        tracing::debug!(
-            %balance_key,
-            ?balance,
-            "Existing value found",
-        );
-        balance.receive(amount);
-        tracing::debug!(
-            %balance_key,
-            ?balance,
-            "New value calculated",
-        );
-    })?;
-    _ = changed_keys.insert(balance_key);
 
-    let supply_key = minted_balance_key(&token);
-    update::amount(wl_storage, &supply_key, |supply| {
-        tracing::debug!(
-            %supply_key,
-            ?supply,
-            "Existing value found",
-        );
-        supply.receive(amount);
-        tracing::debug!(
-            %supply_key,
-            ?supply,
-            "New value calculated",
-        );
-    })?;
-    _ = changed_keys.insert(supply_key);
+    let asset_count = wl_storage
+        .ethbridge_queries()
+        .get_eth_assets_to_mint(asset, amount);
+
+    let assets_to_mint = [
+        // check if we should mint nuts
+        (!asset_count.nut_amount.is_zero())
+            .then(|| (wrapped_erc20s::nut(asset), asset_count.nut_amount)),
+        // check if we should mint erc20s
+        (!asset_count.erc20_amount.is_zero())
+            .then(|| (wrapped_erc20s::token(asset), asset_count.erc20_amount)),
+    ]
+    .into_iter()
+    // remove assets that do not need to be
+    // minted from the iterator
+    .flatten();
+
+    for (token, ref amount) in assets_to_mint {
+        let balance_key = balance_key(&token, receiver);
+        update::amount(wl_storage, &balance_key, |balance| {
+            tracing::debug!(
+                %balance_key,
+                ?balance,
+                "Existing value found",
+            );
+            balance.receive(amount);
+            tracing::debug!(
+                %balance_key,
+                ?balance,
+                "New value calculated",
+            );
+        })?;
+        _ = changed_keys.insert(balance_key);
 
-    let minter_key = minter_key(&token);
-    wl_storage.write_bytes(&minter_key, BRIDGE_POOL_ADDRESS.try_to_vec()?)?;
-    _ = changed_keys.insert(minter_key);
+        let supply_key = minted_balance_key(&token);
+        update::amount(wl_storage, &supply_key, |supply| {
+            tracing::debug!(
+                %supply_key,
+                ?supply,
+                "Existing value found",
+            );
+            supply.receive(amount);
+            tracing::debug!(
+                %supply_key,
+                ?supply,
+                "New value calculated",
+            );
+        })?;
+        _ = changed_keys.insert(supply_key);
 
-    Ok(changed_keys)
+        let minter_key = minter_key(&token);
```

Comment:
what is the purpose of this minter key?

Reply:
this is IBC specific stuff that yuji added for the multitoken PR. it's not actually needed anymore. it basically stores a VP address to check who minted tokens. the ethereum bridge doesn't mint tokens from wasm though. it's already removed in the upstream multitokens PR, but I haven't updated the base yet